### ISSUE TITLE
AudioServer.GetBusEffect causes game to freeze on quit

### ISSUE THREAD

**Issue by mattias800** [godotengine/godot#95404]
### Tested versions

- Reproducible in Godot 4.3 RC3 with C# support.

### System information

Godot v4.3.rc3.mono - macOS 14.5.0 - Vulkan (Forward+) - integrated Apple M1 Ultra - Apple M1 Ultra (20 Threads)

### Issue description

Invoking `AudioServer.GetBusEffect(..)` in the game works fine until I quit the game.
Then the game freezes and I have to force quit it.

This happens in my game, and I can reproduce it in a clean project.

It only happens when using C#. 
When I run identical code in GDScript, the game quits normally and closes immediately.

Log:
```
ERROR: FATAL: Condition "!rc_owner" is true.
```

My installed .NET SDK is 8.0, if that matters.

### Steps to reproduce

1. Use Godot 4.3 RC3 with mono.
2. Create new empty project.
3. Add an audio bus named "Music"
4. Add an insert effect to the bus (low pass filter in my case)
5. Create empty scene.
6. Add a C# script to the scene.
7. Add the following method to the script.
```
    public override void _Ready()
    {
        var l = (AudioEffectLowPassFilter)AudioServer.GetBusEffect(AudioServer.GetBusIndex("Music"), 0);
    }
```
8. Run the scene, then try to close it.
9. It is now frozen and you need to force quit.

### Minimal reproduction project (MRP)

N/A

**Comment by raulsntos** [comment]
- May be a duplicate of https://github.com/godotengine/godot/issues/92076.
- May be fixed by https://github.com/godotengine/godot/pull/93172.

**Comment by mattias800** [comment]
I don't know if it matters, but calling just `AudioServer.GetBusIndex()` does not cause this problem.

**Comment by raulsntos** [comment]
This was likely fixed by https://github.com/godotengine/godot/pull/93172, I forgot to link it. I'm closing this issue, but please let us know if you are still able to reproduce it and it can be reopened. The fix is included in [Godot 4.4-dev.1](https://godotengine.org/article/dev-snapshot-godot-4-4-dev-1/), give it a try and let us know how it goes :slightly_smiling_face:
